staging/src/k8s.io/apimachinery/pkg/api/validate/schema.go

```diff
@@ -72,3 +72,11 @@ func Optional[T comparable](_ operation.Context, fldPath *field.Path, value, _ *
 	}
 	return field.ErrorList{field.Required(fldPath, "optional value was not specified")}
 }
+
+// MaxItems verifies that the specified slice is not longer than max items.
+func MaxItems[T any](_ operation.Context, fldPath *field.Path, value, _ []T, max int) field.ErrorList {
+	if len(value) > max {
```

Comment:
```suggestion
func MaxItems[T any](_ operation.Context, fldPath *field.Path, value, _ []T, max int) field.ErrorList {
        if value == nil {
		return nil
	}
	if len(value) > max {
```

@thockin this is still needed, right? We all validations for absent optional values?

Reply:
If I'm right here, it should be possible to write the corresponding test.  Otherwise ignore me.

Reply:
The `len(nil)` is also 0, so no need to check.

Reply:
Ah, ofcourse. 🤦‍♂️